Automattic/jetpack
# Promote subscription: Add subscription modal on comment

Related to https://github.com/Automattic/dotcom-forge/issues/4584

## Proposed changes:
* Add subscription modal to users after commenting.
* User can disable this feature in `/settings/newsletter/` 
* Calypso PR to enable toggle for Jetpack users: https://github.com/Automattic/wp-calypso/pull/85331
* The mc stat used for this is `jetpack-subscribe-modal-comm`

## To Do
- [x] Subscription is not matching comments toggle (sync issue?)
- [x] We need to check the message origin
- [x] Add cookie to not show > 5 times
- [x] Check if user is already subscribed
- [x] Add custom source for tracking events 
- [x] Some CSS Tweaks
- [x] Limit to Automatticians

### Other information:


## Jetpack product discussion
P2: pb5gDS-3tB-p2

## Does this pull request change what data or activity we track or use?
Not sure



## Testing instructions:
* Follow instructions from here: https://github.com/Automattic/jetpack/pull/34659#issuecomment-1857257562
* Go to a post with a non-site user (admin or editor)
* Comment on the post
* You should see the subscription modal
* Check stats tracking in mc stats

<img width="911" alt="image" src="https://github.com/Automattic/jetpack/assets/402286/a5132c82-9334-41d3-8001-a2f8969175cf">
